cmake/projects/NASM/hunter.cmake

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) 2017, Zhuhao Wang
+# All rights reserved.
+
+# !!! DO NOT PLACE HEADER GUARDS HERE !!!
+
+include(hunter_add_version)
+include(hunter_cacheable)
+include(hunter_download)
+include(hunter_pick_scheme)
+
+if(CMAKE_HOST_WIN32)
+  # NASM only supports a subset of VS compilers on Windows. However, the only
```

Comment:
> So it would be really too much hassle to require user who just want to build some package (e.g., OpenSSL) to install VS (note this is also a commercial software) just to build a program only needed in compiling phase

That's kind of out of scope. I'm just saying if you already using Visual Studio (like in previous version you've send) then it's better to use sources for easy patching reason. And if user want to use MinGW + want to use patched `nasm` + `nasm` can be build only with Visual Studio => not much choice, you have to install Visual Studio, right?

>  Of course it would be great if add_hunter_package can have some parameter to tune if we will build from source or just download pre-build binary

There is binary cache mechanism for it already, if binaries available from server there will be no build triggered.